src/components/Tokens/TokenDetails/TokenDetail.tsx

```diff
@@ -199,12 +206,22 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
     twitterName,
   }))(tokenDetailData)
 
-  if (!token || !token.name || !token.symbol) {
+  if (!token || !token.name || !token.symbol || !connectedChainId) {
     return <LoadingTokenDetail />
   }
 
-  const tokenName = tokenDetailData.name
-  const tokenSymbol = tokenDetailData.tokens?.[0]?.symbol?.toUpperCase() ?? token.symbol
+  const wrappedNativeCurrency = WRAPPED_NATIVE_CURRENCY[connectedChainId]
+  const isWrappedNativeToken = wrappedNativeCurrency && wrappedNativeCurrency.address === token.address
+
+  if (isWrappedNativeToken) {
+    currency = nativeOnChain(connectedChainId)
+  }
+
+  const tokenName = isWrappedNativeToken && currency ? currency.name : tokenDetailData.name
+  const tokenSymbol =
+    isWrappedNativeToken && currency
+      ? currency.symbol
+      : tokenDetailData.tokens?.[0]?.symbol?.toUpperCase() ?? token.symbol
```

Comment:
can we skip the uppercase thing here and move it to css? that will help simplify

Reply:
can you name a variable for `tokenDetailData.tokens?.[0]?.symbol?.toUpperCase() ?? token.symbol` then?